examples/reopen.rs

```diff
@@ -0,0 +1,61 @@
+#![allow(clippy::single_match)]
```

Comment:
I think I'll wait 'till after https://github.com/rust-windowing/winit/pull/3447, as that reworks how our examples are handled.